apps/cowswap-frontend/src/modules/hooksStore/hooks/useBalancesDiff.ts

```diff
@@ -0,0 +1,127 @@
+import { useTenderlyBundleSimulation } from 'modules/tenderly/hooks/useTenderlyBundleSimulation'
+import { useOrderParams } from './useOrderParams'
+import { useHooks } from './useHooks'
+import { useMemo } from 'react'
+import { useWalletInfo } from '@cowprotocol/wallet'
+import { CowHookDetails } from '@cowprotocol/hook-dapp-lib'
+import { BalancesDiff } from 'modules/tenderly/types'
+import { BigNumber } from 'ethers'
+
+const EMPTY_BALANCE_DIFF: BalancesDiff = {}
+
+export function usePreHookBalanceDiff(): BalancesDiff {
+  const { data } = useTenderlyBundleSimulation()
+
+  const { preHooks } = useHooks()
+
+  return useMemo(() => {
+    if (!data || !preHooks.length) return EMPTY_BALANCE_DIFF
+
+    const lastPreHook = preHooks[preHooks.length - 1]
+    return data[lastPreHook?.uuid]?.cumulativeBalancesDiff || EMPTY_BALANCE_DIFF
+  }, [data, preHooks])
+}
+
+// Returns all the ERC20 Balance Diff of the current hook to be passed to the iframe context
+export function useHookBalancesDiff(isPreHook: boolean, hookToEditDetails?: CowHookDetails): BalancesDiff {
+  const { account } = useWalletInfo()
+  const { data } = useTenderlyBundleSimulation()
+  const orderParams = useOrderParams()
+  const { preHooks, postHooks } = useHooks()
+  const preHookBalanceDiff = usePreHookBalanceDiff()
+
+  const orderMockBalanceDiff = useMemo(() => {
+    if (!account) return EMPTY_BALANCE_DIFF
+    const balanceDiff: Record<string, string> = {}
+
+    if (orderParams?.buyAmount && orderParams.buyTokenAddress && account)
+      balanceDiff[orderParams.buyTokenAddress] = orderParams.buyAmount
+
+    if (orderParams?.sellAmount && orderParams.sellTokenAddress && account)
+      balanceDiff[orderParams.sellTokenAddress] = `-${orderParams.sellAmount}`
+
+    return { account: balanceDiff }
+  }, [orderParams, account])
+
+  const firstPostHookBalanceDiff = useMemo(() => {
+    return mergeBalanceDiffs(preHookBalanceDiff, orderMockBalanceDiff)
+  }, [preHookBalanceDiff, orderMockBalanceDiff])
+
+  const postHookBalanceDiff = useMemo(() => {
+    // is adding the first post hook or simulation not available
+    if (!data || !postHooks) return firstPostHookBalanceDiff
+
+    const lastPostHook = postHooks[postHooks.length - 1]
+    return data[lastPostHook?.uuid]?.cumulativeBalancesDiff || firstPostHookBalanceDiff
+  }, [data, postHooks, orderMockBalanceDiff, preHookBalanceDiff])
```

Comment:
`firstPostHookBalanceDiff` is used in the hook but it is not part of the hook deps, while `orderMockBalanceDiff` and `preHookBalanceDiff` are but are not used in the hook.